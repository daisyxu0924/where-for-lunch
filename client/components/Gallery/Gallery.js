import React from 'react';
import PropTypes from 'prop-types';

const Gallery = ({ photos }) => (
  <ul className="gallery">
    {photos.map((photo, i) => (
      <li key={i}>
        <img src={photo} alt={`image-${i + 1}`} />
      </li>
    ))}
  </ul>
);

Gallery.propTypes = {
  photos: PropTypes.arrayOf(PropTypes.string).isRequired,
};

Gallery.defaultProps = {
  photos: [],
};

export default Gallery;
