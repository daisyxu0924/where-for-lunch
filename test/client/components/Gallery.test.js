import React from 'react';
import renderer from 'react-test-renderer';
import Gallery from 'client/components/Gallery/Gallery';

describe('<Gallery />', () => {
  it('should render an empty list if photos property is not supplied.', () => {
    const gallery = renderer
      .create(<Gallery />)
      .toJSON();

    expect(gallery).toMatchSnapshot();
  });

  it('should render a list of photos when photos property is supplied.', () => {
    const photos = [
      'https://s3-media1.fl.yelpcdn.com/bphoto/LmH0H3McZ-WQyf8qAB0h2A/o.jpg',
      'https://s3-media2.fl.yelpcdn.com/bphoto/lbNYdvheIAC2mll5ROvjRw/o.jpg',
      'https://s3-media2.fl.yelpcdn.com/bphoto/dd8oRzkl7VCPpD1d3MV9cw/o.jpg',
    ];
    const gallery = renderer
      .create(<Gallery photos={photos} />)
      .toJSON();

    expect(gallery).toMatchSnapshot();
  });
});
