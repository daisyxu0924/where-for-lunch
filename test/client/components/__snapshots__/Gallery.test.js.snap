// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery /> should render a list of photos when photos property is supplied. 1`] = `
<ul
  className="gallery"
>
  <li>
    <img
      alt="image-1"
      src="https://s3-media1.fl.yelpcdn.com/bphoto/LmH0H3McZ-WQyf8qAB0h2A/o.jpg"
    />
  </li>
  <li>
    <img
      alt="image-2"
      src="https://s3-media2.fl.yelpcdn.com/bphoto/lbNYdvheIAC2mll5ROvjRw/o.jpg"
    />
  </li>
  <li>
    <img
      alt="image-3"
      src="https://s3-media2.fl.yelpcdn.com/bphoto/dd8oRzkl7VCPpD1d3MV9cw/o.jpg"
    />
  </li>
</ul>
`;

exports[`<Gallery /> should render an empty list if photos property is not supplied. 1`] = `
<ul
  className="gallery"
/>
`;
